### PR TITLE
[SYCL] test new complex support by group_algorithms

### PR DESCRIPTION
**Pull request by cperkinsintel**
These test updates support https://github.com/intel/llvm/pull/5394  where we expand some of the group algorithms to support std::complex data type. 

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

File: SYCL/GroupAlgorithm/exclusive_scan_sycl2020.cpp [hunk 1]
```diff
@@ -186,5 +186,16 @@ int main() {
                                 sycl::bit_and<int>(), ~0);
 #endif // SPIRV_1_3
 
+  // as part of SYCL_EXT_ONEAPI_COMPLEX_ALGORITHMS (
+  // https://github.com/intel/llvm/pull/5108/ ) joint_exclusive_scan and
+  // exclusive_scan_over_group now operate on std::complex limited to using the
+  // sycl::plus binary operation.
+  std::array<std::complex<float>, N> input_cf;
+  std::array<std::complex<float>, N> output_cf;
```
Comment (Pennycook) [inline SYCL/GroupAlgorithm/exclusive_scan_sycl2020.cpp:195]: Should we test `float` and `double`?

**Review by Pennycook — CHANGES_REQUESTED** [review]

File: SYCL/GroupAlgorithm/exclusive_scan_sycl2020.cpp (hunk 1 above)
Comment (cperkinsintel) [reply SYCL/GroupAlgorithm/exclusive_scan_sycl2020.cpp:195]: done

**Review by Pennycook — DISMISSED** [review]

**Review by steffenlarsen — COMMENTED** [review]
Tests look good, but I fear it could cause failures on devices without `aspect::fp64`. Do we test on any such devices?

**Comment by cperkinsintel** [comment]
@steffenlarsen  - I added protection to check for aspect::fp64 before testing std::complex<double>

**Comment by steffenlarsen** [comment]
> @steffenlarsen - I added protection to check for aspect::fp64 before testing std::complex

You have! But the problem is that the kernel using fp64 operations may be bundled together with the other kernels. This means when the test kernels are built they may include this kernel and fail.

**Review by vladimirlaz — DISMISSED** [review]

**Comment by vladimirlaz** [comment]
> > @steffenlarsen - I added protection to check for aspect::fp64 before testing std::complex
> 
> You have! But the problem is that the kernel using fp64 operations may be bundled together with the other kernels. This means when the test kernels are built they may include this kernel and fail.

The potential fixes for the situation may be:
 - move fp64 code to separate test;
 - use -fsycl-device-code-split==per_kernel option for building test app.

**Comment by cperkinsintel** [comment]
I believe this is ready for merge.

**Comment by vladimirlaz** [comment]
> I believe this is ready for merge.


don't we want to add  -fsycl-device-code-split=per_kernel to avoid potential failures on devices that do not support fp64?

**Review by steffenlarsen — DISMISSED** [review]
LGTM! Thanks for adding the split. 😄

**Review by steffenlarsen — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**